docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+- Transition to new, improved, K8s API conventions aligned conditions
+
+Below you can find the relevant fields in Machine Status v1beta2, after v1beta1 removal (end state);
+After golang types, you can find a summary table that also shows how changes will be rolled out according to K8s deprecation rules.
+
+```golang
+type MachineDeploymentStatus struct {
+
+    // The number of up-to-date replicas targeted by this deployment.
+    // +optional
+    UpToDateReplicas int32 `json:"upToDateReplicas"`
+
+    // Represents the observations of a MachineDeployment's current state.
+    // +listType=map
+    // +listMapKey=type
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // Other fields...
+    // NOTE: `Phase`, `FailureReason`, `FailureMessage` fields won't be there anymore
+}
+```
+
+| v1beta1 (current)             | v1beta2 (tentative Q1 2025)                              | v1beta1 removal (tentative Q1 2026) |
+|-------------------------------|----------------------------------------------------------|-------------------------------------|
+| `UpdatedReplicas`             | `UpToDateReplicas` (renamed)                             | `UpToDateReplicas`                  |
+| `Phase` (deprecated)          | `BackCompatibilty.Phase` (renamed) (deprecated)          | (removed)                           |
+| `FailureReason` (deprecated)  | `BackCompatibilty.FailureReason` (renamed) (deprecated)  | (removed)                           |
+| `FailureMessage` (deprecated) | `BackCompatibilty.FailureMessage` (renamed) (deprecated) | (removed)                           |
+| `Conditions`                  | `BackCompatibilty.Conditions` (renamed) (deprecated)     | (removed)                           |
+| `ExprimentalConditions` (new) | `Conditions` (renamed)                                   | `Conditions`                        |
+| other fields...               | other fields...                                          | other fields...                     |
+
+Notes:
+- The `BackCompatibilty` struct is going to exist in v1beta2 types only until v1beta1 removal (9 months or 3 minor releases after v1beta2 is released/v1beta1 is deprecated, whichever is longer).
+  Fields in this struct are used for supporting down conversions, thus providing users relying on v1beta1 APIs additional buffer time to pick up the new changes.
+
+#### MachineDeployment (New)Conditions
+
+| Condition        | Note                                                                                                                                                                                                                                                   |
+|------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```

Comment:
From a consumer pov it'd be valuable to look at a condition and understand if may MachineDeployment is performing a rolling upgrade and it was triggered by an infraMachineTemplate change or a kube version change

Reply:
Agreed, we already did some work to surface the reason why a rollout happens il logs, the next logical step is to have this in conditions too, but there is still some work to go from the raw diff we have today to a simple summary like the one you suggested.

Reply:
Resolving old comments without feedback.
Also added as a point in https://github.com/kubernetes-sigs/cluster-api/issues/11105